### ISSUE TITLE
--strip should emit no debug info and result in a stripped binary according to binutils

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#351]
`--strip` is supposed to not emit debug info, but it still does.

Additionally if you use `file` on the binary it should say "stripped" and it should not say "not stripped".

**Comment by andrewrk** [comment]
Postponing to 0.3.0. We only have to solve this issue in the self-hosted compiler.

**Comment by andrewrk** [comment]
Strip now works fairly well but it does not strip everything. It's no longer a bug.

**Comment by artob** [comment]
This bit me just now when packaging my project for Debian. Namely, the last stage (linting) of `debuild` shows an error due to the shared libraries produced by Zig 0.6.0 with `--strip` still indicating that they are unstripped in the ELF information (scroll to the end of the output from `file`):

```bash
$ zig build-lib --name foo -target x86_64-linux-gnu -dynamic --strip

$ file libfoo.so.0.0.0
libfoo.so.0.0.0: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, not stripped

$ strip libfoo.so.0.0.0

$ file libfoo.so.0.0.0
libfoo.so.0.0.0: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, stripped
```

The [Lintian](https://manpages.debian.org/stable/lintian/lintian.1.en.html) error is [unstripped-binary-or-object](https://lintian.debian.org/tags/unstripped-binary-or-object.html):

```
E: libfoo0: unstripped-binary-or-object usr/lib/x86_64-linux-gnu/libfoo.so.0.0.0
N: 
N:    The package installs an unstripped binary or object file.
N:    
N:    Please note, that shared libraries have to be stripped with the
N:    --strip-unneeded option.
N:    
N:    Refer to Debian Policy Manual section 10.1 (Binaries) and Debian Policy
N:    Manual section 10.2 (Libraries) for details.
N:    
N:    Severity: important, Certainty: certain
N:    
N:    Check: binaries, Type: binary, udeb
```

This will affect all shared libraries produced by Zig when packaged for Debian or Ubuntu. The workaround for now is to either add an explicit spurious `strip(1)` invocation after Zig builds the library, or else add a Lintian override to suppress the error.

**Comment by Ristovski** [comment]
This appears to be fixed. (By https://github.com/ziglang/zig/commit/7a0022725342be8d3ea555a635da51f0e8c47b25 I assume?)

```
$ zig build-exe test.zig -femit-bin=striptest
$ file striptest
striptest: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, with debug_info, not stripped
```
```
$ zig build-exe test.zig -femit-bin=striptest --strip
$ file striptest
striptest: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, stripped
```

0.10.0-dev.2562+61844b6bd

**Comment by andrewrk** [comment]
Fixed by 366c76744429cb9c2fcd60abad191b7ef40ed5db
